### DISCUSSION TITLE
Using NFID locally without `ngrok`

### DISCUSSION THREAD

**Discussion by paulyoung** [internet-identity-labs/NFID-SDK#143] [Q&A]
I'd rather not use `ngrok`. Is there any way around that?

I thought the QR code used to be clickable on desktop but that isn't working now.

Perhaps there could be a preconfigured test user whose recovery phrase is documented so we can easily recover the account.

**Comment by paulyoung** [comment]
1. Scan QR code with phone
1. Copy URL from phone browser
    * In the format http://localhost:9090/rdp/c003f48c-c432-4f13-bca5-5a9c82df5911/localhost:8080?applicationName=Your%20Application%20Name&applicationLogo=http%253A%252F%252F127.0.0.1%253A8080%252Flogo-37.png
1. Paste URL into desktop browser
1. Create NFID

**Reply by dostro** [reply]
You could also use a browser extension like [QR code reader](https://chrome.google.com/webstore/detail/qr-code-reader/likadllkkidlligfcdhfnnbkjigdkmci/) to:
1. Scan QR code with browser extension
2. Copy/paste URL into desktop browser
3. Create NFID

**Reply by paulyoung** [reply]
I'm running into problems when using this approach, and @dostro said it's not a supported workflow, so I'm still looking for alternatives.

**Reply by paulyoung** [reply]
I gave in and tried `ngrok` but ran into https://github.com/inconshreveable/ngrok/issues/649 / https://github.com/inconshreveable/ngrok/issues/653 which have been open for 2 years...

**Reply by paulyoung** [reply]
I encountered the issues about when downloading `ngrok` from their website but `npm install ngrok` seems to work so I'm trying that.

**Comment by paulyoung** [comment]
I'm currently trying this and it seems promising:

```js
const AUTH_PATH = `/authenticate/?applicationName=${APPLICATION_NAME}&applicationLogo=${APPLICATION_LOGO_URL}#authorize`;
const NFID_URL = DFX_NETWORK === "ic" ? "https://nfid.one" : "http://<ip address of my computer on local network>:9090";
const NFID_AUTH_URL = `${NFID_URL}${AUTH_PATH}`
```

**Reply by paulyoung** [reply]
The issue I'm running into now is that after scanning the QR code on my phone I'm taken to a URL that starts with `/rdp/...` and my browser says `Cannot GET /rdp/...`

**Reply by paulyoung** [reply]
Apparently this won't work because HTTPS is required.
